app/src/main/java/fr/free/nrw/commons/LocationPicker/LocationPickerActivity.java

```diff
@@ -452,11 +460,51 @@ private void addCenterOnGPSButton(){
         fabCenterOnLocation = findViewById(R.id.center_on_gps);
         fabCenterOnLocation.setOnClickListener(view -> getCenter());
     }
+
     /**
-     * Animate map to move to desired Latitude and Longitude
+     * Center the map at user's current location
      */
-    void getCenter() {
-        mapboxMap.animateCamera(CameraUpdateFactory.newLatLngZoom(new LatLng(location.getLatitude(),location.getLongitude()),15.0));
+    private void getCenter() {
+        fr.free.nrw.commons.location.LatLng currLocation = locationManager.getLastLocation();
+        if (currLocation != null) {
+            mapboxMap.animateCamera(CameraUpdateFactory.newLatLngZoom(
+                    new LatLng(currLocation.getLatitude(), currLocation.getLongitude()), 15.0));
+        } else if (locationManager.isGPSProviderEnabled() ||
+            locationManager.isNetworkProviderEnabled()) {
+            locationManager.requestLocationUpdatesFromProvider(LocationManager.NETWORK_PROVIDER);
+            locationManager.requestLocationUpdatesFromProvider(LocationManager.GPS_PROVIDER);
```

Comment:
The location does not seem updated based on the code in this else block which is confusing. Could you kindly elaborate what are you trying to achieve here? 🤔

Reply:
I took three cases: one where the last location is known already(recently updated case), another where we are able to request for updates from the GPS/Network Provider and the third where we don't have location access at all. I was trying to replicate the behaviour exhibited in the Nearby fragment since it does not crash on tapping the target icon and the code seemed promising, so that all the map fragments have the same behaviour for this icon. However, I just noticed nearby does not follow the flow you have described above. In fact, it does not ask for the location access at all and simply points to a predefined location:

https://user-images.githubusercontent.com/83745993/232448245-c236b31d-c3fe-45d2-9cc8-fdf5cde71366.mp4


The map in the Explore tab seems to work as intended, though:

https://user-images.githubusercontent.com/83745993/232448549-b789f00d-215f-4aed-850c-53177848345a.mp4

I think I need to understand a bit more about the different map fragments.